### DISCUSSION TITLE
Call for hackathons for the 2024 CF workshop

### DISCUSSION THREAD

**Discussion by davidhassell** [cf-convention/discuss#342] [Announcements relevant to CF]
Hello,

We are still looking for some suggestions for hackathons in the [2024 CF meeting](https://cfconventions.org/Meetings/2024-Workshop.html#preliminary-agenda). The hackathons will comprise "part II" of the meeting on 19-20 September and will follow the same format as previous meetings, i.e. parallel breakout groups will each  discuss a specific issue with the aim of progressing an enhancement to CF, where "progressing" can mean anything from simply increasing understanding of a topic, to writing a pull request to an already agreed issue.

The [meeting agenda](https://cfconventions.org/Meetings/2024-Workshop.html#preliminary-agenda) meeting has three hackathon sessions, each of which will contain 3 or 4 hackathons in parallel - the idea being that it's then likely that there's something of interest for everyone in each session.

The only pre-requisite for suggesting a hackathon is a commitment to provide (by the time of the meeting) a short paragraph introducing the topic, and to lead (or at least initiate) the discussion during the breakout session. The hackathons are meant to be a mainly group-based activity, so no presentation is necessary, unless you want to show a few slides by way of an introduction.    

Any subject can be considered for a hackathon, but we have found in the past that ongoing GitHub issues can provide good topics. To that end, I have collated below a list of open CF enhancement issues (pinging their originator) in case that provides anyone with inspiration.

If you would like to suggest a hackathon topic, please do reply to this discussion.

_(If you have already suggested one on through other channels, it won't harm to mention it again here, for completeness!)_

Many thanks,
David, for the organising committee 

### Discussions

- [Composite/array values in string valued global attributes](https://github.com/orgs/cf-convention/discussions/341) @benjwadams started Jul 17, 2024

- [Strengthen "Groups" conventions](https://github.com/orgs/cf-convention/discussions/333) @pvanlaake started Jul 8, 2024 in Comments and ideas for changing CF

- [Allow user-defined types as a data type](https://github.com/orgs/cf-convention/discussions/331) @pvanlaake started Jun 30, 2024

- [Concluding discussions](https://github.com/orgs/cf-convention/discussions/313) @JonathanGregory started May 10, 20248

- [Creating a blacklisting certain characters from variable and attribute names](https://github.com/orgs/cf-convention/discussions/323) @larsbarring started May 31, 2024

- [apparent contradiction about recommendation of standard_name or long_name](https://github.com/orgs/cf-convention/discussions/300) @JonathanGregory started Mar 20, 2024

- [How to do vector fields under CF?](https://github.com/orgs/cf-convention/discussions/318) @twhitehead asked May 18, 20242

- [How to deal with standard names having a <space> character](https://github.com/orgs/cf-convention/discussions/310) @larsbarring started May 7, 2024

- [Can CF offer better support for the representation of climate anomalies?](https://github.com/orgs/cf-convention/discussions/305) @TomLav started Apr 12, 2024

- [UTC leap-seconds (again!)](https://github.com/orgs/cf-convention/discussions/304) @ChrisBarker-NOAA started Mar 22, 2024

### cf-convention/discuss/issues

- [Question about day of year climatology statistics](https://github.com/cf-convention/discuss/issues/301) @huaracheguarache

- [Clarification of time stamps](https://github.com/cf-convention/discuss/issues/297) opened Mar 15, 2024 by @claashk

- [Introduce global attributes "license" and "copyright"?](https://github.com/cf-convention/discuss/issues/294) opened Mar 11, 2024 by @rjhogan

- [How can we encode a DSG trajectory has two identifiers?](https://github.com/cf-convention/discuss/issues/282) opened Mar 1, 2024 by @davidhassell

- [Order of dimensions for ensemble model output](https://github.com/cf-convention/discuss/issues/275) opened Jan 24, 2024 by @knutfrode

- [cell_methods for covariance](https://github.com/cf-convention/discuss/issues/269) opened Jan 2, 2024 by @achho

- [OGC GeoZarr working group: Comments sought](https://github.com/cf-convention/discuss/issues/259) opened Aug 25, 2023 by @erget

- [Question about ship data](https://github.com/cf-convention/discuss/issues/257) opened Aug 2, 2023 by @huaracheguarache

- [Instantaneous variables](https://github.com/cf-convention/discuss/issues/250) opened Jul 3, 2023 by @MargaridaS

- [Time series of profiles with multiple modes](https://github.com/cf-convention/discuss/issues/249) opened Jun 29, 2023 by @donmurray

- [Localized metadata in NetCDF files](https://github.com/cf-convention/discuss/issues/244) opened Jun 1, 2023 by @turnbullerin

- [Multiple Feature Types in one CF file](https://github.com/cf-convention/discuss/issues/231) opened May 3, 2023 by @gaochen-larc

- [Values for variable sector](https://github.com/cf-convention/discuss/issues/227) opened Apr 25, 2023 by @MargaridaS

- [Example Datasets for Section 8.3: Lossy Compression](https://github.com/cf-convention/discuss/issues/200) opened Dec 23, 2022 by @benjwadams

- [Any convention on what flag_meaning to use to QC pass?](https://github.com/cf-convention/discuss/issues/184) opened Oct 5, 2022 by @David-Rayner-GVC

- [Nesting of ancillary_variables attribute?](https://github.com/cf-convention/discuss/issues/174) opened Aug 31, 2022 by @frankeye

- [Use of "where" in cell_methods](https://github.com/cf-convention/discuss/issues/173) opened Aug 16, 2022 by @taylor13

- [Issue with time variable for Monte Carlo "scenarios"](https://github.com/cf-convention/discuss/issues/164) opened Jul 8, 2022 by @danielbwright

- [For Info: netCDF linked data](https://github.com/cf-convention/discuss/issues/160) opened Jun 1, 2022 by @marqh

- [Is scalar flag_masks allowed?](https://github.com/cf-convention/discuss/issues/151) opened Apr 6, 2022 by @jdemaria

- ["area: mean where type1 over type2"](https://github.com/cf-convention/discuss/issues/149) opened Mar 29, 2022 by @taylor13

- [Use of default fill value](https://github.com/cf-convention/discuss/issues/145) opened Feb 25, 2022 by @timhultberg

- [CF convention update from 1.6 to 1.8 and ACDD alignment](https://github.com/cf-convention/discuss/issues/116) opened Jul 6, 2021 by @Aptychus

- [String/Char labels (section 6) mismatch between Conformance and CF Document](https://github.com/cf-convention/discuss/issues/111) opened Jun 1, 2021 by @DocOtak

### cf-convention/cf-conventions/issues

- [Conventions for placing CF constructs in a group hierarchy](https://github.com/cf-convention/cf-conventions/issues/533) opened Jul 29, 2024 by @pvanlaake

- [Support localization of natural language attributes and variables](https://github.com/cf-convention/cf-conventions/issues/528) opened Jul 8, 2024 by @turnbullerin

- [Include DOI and License information in the conventions document](https://github.com/cf-convention/cf-conventions/issues/513) opened Mar 11, 2024 by @larsbarring

- [Incorporating the CFA convention for aggregated datasets into CF](https://github.com/cf-convention/cf-conventions/issues/508) opened Feb 7, 2024 by @davidhassell
- [Clarification of weighting in cell_methods](https://github.com/cf-convention/cf-conventions/issues/447) opened Aug 22, 2023 by @JonathanGregory

- [Convention for HEALPix grid parameters](https://github.com/cf-convention/cf-conventions/issues/433) opened Mar 18, 2023 by @uweschulzweida

- [Clarification of cell_methods](https://github.com/cf-convention/cf-conventions/issues/414) opened Nov 23, 2022 by @taylor13

- [Color blindness accessibility of document figures & tables](https://github.com/cf-convention/cf-conventions/issues/404) opened Sep 27, 2022 by @sadielbartholomew

- [Metadata to encode quantization properties](https://github.com/cf-convention/cf-conventions/issues/403) opened Sep 15, 2022 by @czender

- [Mapping between standard names and GRIB codes](https://github.com/cf-convention/cf-conventions/issues/367) opened May 12, 2022 by @atmodatcode

- [Location / Identification for grid specs outside of a file](https://github.com/cf-convention/cf-conventions/issues/357) opened Mar 10, 2022 by @ChrisBarker-NOAA

- [Provision of netCDF files for the netCDF header examples shown in the Conventions?](https://github.com/cf-convention/cf-conventions/issues/348) opened Jan 10, 2022 by @atmodatcode

- [Support Swath Data in CF](https://github.com/cf-convention/cf-conventions/issues/269) opened May 14, 2020 by @ajelenak

- [Proposal: Example for Independent Latitude, Longitude, Non-spatiotemporal Variable, and Time Axes](https://github.com/cf-convention/cf-conventions/issues/249) opened Mar 2, 2020 by @GeyerB

- [Clarification of time coordinate requirements](https://github.com/cf-convention/cf-conventions/issues/166) opened Jun 20, 2019 by @martinjuckes

- [Add support for attributes of type string](https://github.com/cf-convention/cf-conventions/issues/141) opened Jul 23, 2018 by @JimBiardCics

**Comment by sadielbartholomew** [comment]
Hi David,

I put this on my sign-up form, so you and the organising team may already be aware (but you did say it wouldn't harm to mention again): I would really like to get some code I made years ago towards #cf-convention/cf-convention.github.io/issues/110 updated, tidied and added to the repo so that we can generate and display on the website the plots of number of standard names over time and pre-release plus word clouds of any given set e.g. those added between one release and the next. This would work well as a hackathon topic I think - and without dedicating some time e.g. through a hackathon session, I doubt I'd find time to do that myself.

Any questions please let me know but I think it should be fairly self-explanatory given the Issue and the context above of me having made code years ago that needs updating and then we can add it as a script in one of our repos with instructions to generate the new plots, and adding them on the site to cover the latest versions.  The code or scope fro visualisations could be extended too, for anyone with ideas in that respect.

Thanks.

**Comment by gtesoro** [comment]
Hi David,

My name is Guillermo Tesoro and I will be giving a lighting presentation on a work-in-progress general purpose netCDF Schema on day  2. The idea is to have a machine and human readable (yaml) schema definition that can be use to validate any netCDF file. The initial goal post of the project is to have all the features necessary to cover the full CF Conventions. On the coming days, I will open an issue on the repo with further details so we can already start dicussing.

I think it will be great to have a follow up session during the hackathon to contribute ideas and share critizisms to the schema.

Cheers

**Comment by ChrisBarker-NOAA** [comment]
2024 workshop passed -- closing.
